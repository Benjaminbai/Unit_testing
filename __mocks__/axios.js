
const mockAxios = jest.requireActual('Axios')

export default mockAxios;