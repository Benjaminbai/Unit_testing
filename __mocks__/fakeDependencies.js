jest.mock('react-router-dom', () => {
    const originalModule = jest.requireActual('react-router-dom');

    return {
        __esModule: true,
        ...originalModule,
        useLocation: jest.fn().mockReturnValue({
            pathname: '',
            search: '',
            hash: '',
            state: null,
            key: '',
        }),
        useHistory: () => ({
            push: jest.fn(),
            location: {}
        }),
        Redirect: jest.fn().mockReturnValue({}),
    }
})
window.matchMedia = jest.fn().mockImplementation((query) => {
    const obj = {
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(),
        removeListener: jest.fn(),
    };

    return obj;
})

jest.mock('react', () => ({
    ...jest.requireActual('react'),
    useLayoutEffect: jest.requireActual('react').useEffect
}))