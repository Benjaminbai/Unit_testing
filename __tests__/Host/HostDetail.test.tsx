import '../../__mocks__/fakeDependencies'
import React from 'react'
import { mount, render } from 'enzyme'
import HostDetail from '../../pages/Host/HostDetail'
import { BrowserRouter as Router } from 'react-router-dom'

import { render as RTLRender } from '@testing-library/react'
import API from '../../api/Host'

describe('<HostDetail />', () => {
    test('expect the HostDetail exist', () => {
        const HostDetailComponent = mount(<Router><HostDetail /></Router>)
        expect(HostDetailComponent.exists()).toBe(true)
    })

    test('expect HostDetail unmount', () => {
        const HostDetailComponent = mount(<Router><HostDetail /></Router>)
        HostDetailComponent.unmount()
        expect(() => {
            HostDetailComponent.instance()
        }).toThrow()
    })

    // 快照
    test('the source code changed or not', () => {
        const HostDetailComponent = render(<Router><HostDetail /></Router>)
        expect(HostDetailComponent).toMatchSnapshot()
    })

    test('the count of tab should be five', () => {
        const HostDetailComponent = mount(<Router><HostDetail /></Router>)
        expect(HostDetailComponent.find('.ant-tabs-tab').length).toBe(5)
    })

    test('chart rendered', () => {
        const HostDetailComponent = mount(<Router><HostDetail /></Router>)
        expect(HostDetailComponent.find('DiskIO').exists()).toBe(true)
    })

    test('detailRegistryHost api', () => {
        const spyFn = jest.spyOn(API, 'detailRegistryHost')
        RTLRender(<Router><HostDetail /></Router>)
        expect(spyFn).toBeCalled()
    })
})