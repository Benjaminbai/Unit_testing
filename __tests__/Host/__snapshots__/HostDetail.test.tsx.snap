// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HostDetail /> the source code changed or not 1`] = `
<section
  class="ant-layout layout"
  style="height:100%"
>
  <header
    class="ant-layout-header header"
    style="background-color:#ffffff"
  >
    <div
      class="logo"
      style="margin:16px 0;width:200px"
    >
      <img
        alt="logo"
        width="120px"
      />
    </div>
    <div
      class=""
    >
      <ul
        class="ant-menu ant-menu-light right-container ant-menu-root ant-menu-horizontal"
        role="menu"
        style="float:right"
      >
        <li
          class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
          role="menuitem"
          style="display:none"
        >
          <div
            aria-expanded="false"
            aria-haspopup="true"
            class="ant-menu-submenu-title"
            role="button"
          >
            <span>
              ···
            </span>
            <i
              class="ant-menu-submenu-arrow"
            />
          </div>
        </li>
        <li
          class="ant-menu-submenu ant-menu-submenu-horizontal"
          role="menuitem"
        >
          <div
            aria-expanded="false"
            aria-haspopup="true"
            class="ant-menu-submenu-title"
            role="button"
          >
            <div>
              <span
                aria-label="user"
                class="anticon anticon-user"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="user"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
                  />
                </svg>
              </span>
              -
            </div>
            <i
              class="ant-menu-submenu-arrow"
            />
          </div>
        </li>
        <li
          class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
          role="menuitem"
          style="visibility:hidden;position:absolute"
        >
          <div
            aria-expanded="false"
            aria-haspopup="true"
            class="ant-menu-submenu-title"
            role="button"
          >
            <span>
              ···
            </span>
            <i
              class="ant-menu-submenu-arrow"
            />
          </div>
        </li>
      </ul>
    </div>
  </header>
  <section
    class="ant-layout"
  >
    <aside
      class="ant-layout-sider ant-layout-sider-dark site-layout-background"
      style="flex:0 0 200px;max-width:200px;min-width:200px;width:200px"
    >
      <div
        class="ant-layout-sider-children"
      >
        <ul
          class="ant-menu ant-menu-light ant-menu-root ant-menu-inline"
          role="menu"
          style="height:100%;border-right:0"
        >
          <li
            class="ant-menu-item"
            role="menuitem"
            style="padding-left:24px"
          >
            <span
              aria-label="home"
              class="anticon anticon-home"
              role="img"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="home"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M946.5 505L560.1 118.8l-25.9-25.9a31.5 31.5 0 00-44.4 0L77.5 505a63.9 63.9 0 00-18.8 46c.4 35.2 29.7 63.3 64.9 63.3h42.5V940h691.8V614.3h43.4c17.1 0 33.2-6.7 45.3-18.8a63.6 63.6 0 0018.7-45.3c0-17-6.7-33.1-18.8-45.2zM568 868H456V664h112v204zm217.9-325.7V868H632V640c0-22.1-17.9-40-40-40H432c-22.1 0-40 17.9-40 40v228H238.1V542.3h-96l370-369.7 23.1 23.1L882 542.3h-96.1z"
                />
              </svg>
            </span>
            <span>
              <a
                href="/dashboard"
              >
                Dashboard
              </a>
            </span>
          </li>
          <li
            class="ant-menu-submenu ant-menu-submenu-inline"
            role="menuitem"
          >
            <div
              aria-expanded="false"
              aria-haspopup="true"
              class="ant-menu-submenu-title"
              role="button"
              style="padding-left:24px"
            >
              <span
                aria-label="aim"
                class="anticon anticon-aim"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="aim"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <defs />
                  <path
                    d="M952 474H829.8C812.5 327.6 696.4 211.5 550 194.2V72c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v122.2C327.6 211.5 211.5 327.6 194.2 474H72c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h122.2C211.5 696.4 327.6 812.5 474 829.8V952c0 4.4 3.6 8 8 8h60c4.4 0 8-3.6 8-8V829.8C696.4 812.5 812.5 696.4 829.8 550H952c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8zM512 756c-134.8 0-244-109.2-244-244s109.2-244 244-244 244 109.2 244 244-109.2 244-244 244z"
                  />
                  <path
                    d="M512 392c-32.1 0-62.1 12.4-84.8 35.2-22.7 22.7-35.2 52.7-35.2 84.8s12.5 62.1 35.2 84.8C449.9 619.4 480 632 512 632s62.1-12.5 84.8-35.2C619.4 574.1 632 544 632 512s-12.5-62.1-35.2-84.8A118.57 118.57 0 00512 392z"
                  />
                </svg>
              </span>
              <span>
                Agent
              </span>
              <i
                class="ant-menu-submenu-arrow"
              />
            </div>
            <div />
          </li>
          <li
            class="ant-menu-submenu ant-menu-submenu-inline"
            role="menuitem"
          >
            <div
              aria-expanded="false"
              aria-haspopup="true"
              class="ant-menu-submenu-title"
              role="button"
              style="padding-left:24px"
            >
              <span
                aria-label="deployment-unit"
                class="anticon anticon-deployment-unit"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="deployment-unit"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M888.3 693.2c-42.5-24.6-94.3-18-129.2 12.8l-53-30.7V523.6c0-15.7-8.4-30.3-22-38.1l-136-78.3v-67.1c44.2-15 76-56.8 76-106.1 0-61.9-50.1-112-112-112s-112 50.1-112 112c0 49.3 31.8 91.1 76 106.1v67.1l-136 78.3c-13.6 7.8-22 22.4-22 38.1v151.6l-53 30.7c-34.9-30.8-86.8-37.4-129.2-12.8-53.5 31-71.7 99.4-41 152.9 30.8 53.5 98.9 71.9 152.2 41 42.5-24.6 62.7-73 53.6-118.8l48.7-28.3 140.6 81c6.8 3.9 14.4 5.9 22 5.9s15.2-2 22-5.9L674.5 740l48.7 28.3c-9.1 45.7 11.2 94.2 53.6 118.8 53.3 30.9 121.5 12.6 152.2-41 30.8-53.6 12.6-122-40.7-152.9zm-673 138.4a47.6 47.6 0 01-65.2-17.6c-13.2-22.9-5.4-52.3 17.5-65.5a47.6 47.6 0 0165.2 17.6c13.2 22.9 5.4 52.3-17.5 65.5zM522 463.8zM464 234a48.01 48.01 0 0196 0 48.01 48.01 0 01-96 0zm170 446.2l-122 70.3-122-70.3V539.8l122-70.3 122 70.3v140.4zm239.9 133.9c-13.2 22.9-42.4 30.8-65.2 17.6-22.8-13.2-30.7-42.6-17.5-65.5s42.4-30.8 65.2-17.6c22.9 13.2 30.7 42.5 17.5 65.5z"
                  />
                </svg>
              </span>
              <span>
                Master
              </span>
              <i
                class="ant-menu-submenu-arrow"
              />
            </div>
            <div />
          </li>
          <li
            class="ant-menu-submenu ant-menu-submenu-inline"
            role="menuitem"
          >
            <div
              aria-expanded="false"
              aria-haspopup="true"
              class="ant-menu-submenu-title"
              role="button"
              style="padding-left:24px"
            >
              <span
                aria-label="tags"
                class="anticon anticon-tags"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="tags"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M483.2 790.3L861.4 412c1.7-1.7 2.5-4 2.3-6.3l-25.5-301.4c-.7-7.8-6.8-13.9-14.6-14.6L522.2 64.3c-2.3-.2-4.7.6-6.3 2.3L137.7 444.8a8.03 8.03 0 000 11.3l334.2 334.2c3.1 3.2 8.2 3.2 11.3 0zm62.6-651.7l224.6 19 19 224.6L477.5 694 233.9 450.5l311.9-311.9zm60.16 186.23a48 48 0 1067.88-67.89 48 48 0 10-67.88 67.89zM889.7 539.8l-39.6-39.5a8.03 8.03 0 00-11.3 0l-362 361.3-237.6-237a8.03 8.03 0 00-11.3 0l-39.6 39.5a8.03 8.03 0 000 11.3l243.2 242.8 39.6 39.5c3.1 3.1 8.2 3.1 11.3 0l407.3-406.6c3.1-3.1 3.1-8.2 0-11.3z"
                  />
                </svg>
              </span>
              <span>
                版本管理
              </span>
              <i
                class="ant-menu-submenu-arrow"
              />
            </div>
            <div />
          </li>
          <li
            class="ant-menu-submenu ant-menu-submenu-inline"
            role="menuitem"
          >
            <div
              aria-expanded="false"
              aria-haspopup="true"
              class="ant-menu-submenu-title"
              role="button"
              style="padding-left:24px"
            >
              <span
                aria-label="desktop"
                class="anticon anticon-desktop"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="desktop"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M928 140H96c-17.7 0-32 14.3-32 32v496c0 17.7 14.3 32 32 32h380v112H304c-8.8 0-16 7.2-16 16v48c0 4.4 3.6 8 8 8h432c4.4 0 8-3.6 8-8v-48c0-8.8-7.2-16-16-16H548V700h380c17.7 0 32-14.3 32-32V172c0-17.7-14.3-32-32-32zm-40 488H136V212h752v416z"
                  />
                </svg>
              </span>
              <span>
                主机信息
              </span>
              <i
                class="ant-menu-submenu-arrow"
              />
            </div>
            <div />
          </li>
        </ul>
      </div>
    </aside>
    <section
      class="ant-layout"
      style="padding:24px"
    >
      <div
        class="ant-breadcrumb"
        style="margin-bottom:16px"
      >
        <span>
          <span
            class="ant-breadcrumb-link"
          >
            主机信息
          </span>
          <span
            class="ant-breadcrumb-separator"
          >
            /
          </span>
        </span>
        <span>
          <span
            class="ant-breadcrumb-link"
          >
            <a
              href="/host/list"
            >
              主机列表
            </a>
          </span>
          <span
            class="ant-breadcrumb-separator"
          >
            /
          </span>
        </span>
      </div>
      <main
        class="ant-layout-content site-layout-background"
        style="padding:20px;margin:0;min-height:280px"
      >
        <div
          style="height:100%;overflow-y:scroll;overflow-x:hidden"
        >
          <h3
            style="border-bottom:1px solid #ececec;padding:10px"
          >
            <span
              aria-label="laptop"
              class="anticon anticon-laptop"
              role="img"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="laptop"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M956.9 845.1L896.4 632V168c0-17.7-14.3-32-32-32h-704c-17.7 0-32 14.3-32 32v464L67.9 845.1C60.4 866 75.8 888 98 888h828.8c22.2 0 37.6-22 30.1-42.9zM200.4 208h624v395h-624V208zm228.3 608l8.1-37h150.3l8.1 37H428.7zm224 0l-19.1-86.7c-.8-3.7-4.1-6.3-7.8-6.3H398.2c-3.8 0-7 2.6-7.8 6.3L371.3 816H151l42.3-149h638.2l42.3 149H652.7z"
                />
              </svg>
            </span>
             系统信息
          </h3>
          <div
            style="padding:0 20px 10px 20px"
          >
            <div
              class="ant-row"
              style="margin-left:-8px;margin-right:-8px;margin-bottom:10px"
            >
              <div
                class="ant-col ant-col-lg-3 ant-col-xxl-2"
                style="padding-left:8px;padding-right:8px"
              >
                <strong>
                  主机名：
                </strong>
              </div>
              <div
                class="ant-col ant-col-lg-3 ant-col-xxl-4"
                style="padding-left:8px;padding-right:8px;word-break:break-word"
              />
              <div
                class="ant-col ant-col-lg-3 ant-col-xxl-2"
                style="padding-left:8px;padding-right:8px"
              >
                <strong>
                  操作系统：
                </strong>
              </div>
              <div
                class="ant-col ant-col-lg-3 ant-col-xxl-4"
                style="padding-left:8px;padding-right:8px"
              />
              <div
                class="ant-col ant-col-lg-3 ant-col-xxl-2"
                style="padding-left:8px;padding-right:8px"
              >
                <strong>
                  发行版：
                </strong>
              </div>
              <div
                class="ant-col ant-col-lg-3 ant-col-xxl-4"
                style="padding-left:8px;padding-right:8px"
              >
                 
              </div>
            </div>
            <div
              class="ant-row"
              style="margin-left:-8px;margin-right:-8px;margin-bottom:10px"
            >
              <div
                class="ant-col ant-col-lg-3 ant-col-xxl-2"
                style="padding-left:8px;padding-right:8px"
              >
                <strong>
                  内核版本：
                </strong>
              </div>
              <div
                class="ant-col ant-col-lg-3 ant-col-xxl-4"
                style="padding-left:8px;padding-right:8px"
              />
              <div
                class="ant-col ant-col-lg-3 ant-col-xxl-2"
                style="padding-left:8px;padding-right:8px"
              >
                <strong>
                  内核架构：
                </strong>
              </div>
              <div
                class="ant-col ant-col-lg-3 ant-col-xxl-4"
                style="padding-left:8px;padding-right:8px"
              />
              <div
                class="ant-col ant-col-lg-3 ant-col-xxl-2"
                style="padding-left:8px;padding-right:8px"
              >
                <strong>
                  运行时间:
                </strong>
              </div>
              <div
                class="ant-col ant-col-lg-3 ant-col-xxl-4"
                style="padding-left:8px;padding-right:8px"
              />
            </div>
          </div>
          <h3
            style="border-bottom:1px solid #ececec;padding:10px"
          >
            <span
              aria-label="hdd"
              class="anticon anticon-hdd"
              role="img"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="hdd"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M832 64H192c-17.7 0-32 14.3-32 32v832c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V96c0-17.7-14.3-32-32-32zm-600 72h560v208H232V136zm560 480H232V408h560v208zm0 272H232V680h560v208zM496 208H312c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h184c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8zM312 544h184c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8H312c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8zm328 244a40 40 0 1080 0 40 40 0 10-80 0z"
                />
              </svg>
            </span>
             硬件信息
          </h3>
          <div
            style="padding:0 20px 10px 20px"
          >
            <div
              class="ant-row"
              style="margin-left:-8px;margin-right:-8px;margin-bottom:10px"
            >
              <div
                class="ant-col ant-col-lg-3 ant-col-xxl-2"
                style="padding-left:8px;padding-right:8px"
              >
                <strong>
                  SN：
                </strong>
              </div>
              <div
                class="ant-col ant-col-lg-3 ant-col-xxl-4"
                style="padding-left:8px;padding-right:8px;word-break:break-word"
              />
              <div
                class="ant-col ant-col-lg-3 ant-col-xxl-2"
                style="padding-left:8px;padding-right:8px"
              >
                <strong>
                  发行商：
                </strong>
              </div>
              <div
                class="ant-col ant-col-lg-3 ant-col-xxl-4"
                style="padding-left:8px;padding-right:8px"
              />
              <div
                class="ant-col ant-col-lg-3 ant-col-xxl-2"
                style="padding-left:8px;padding-right:8px"
              >
                <strong>
                  机器类型：
                </strong>
              </div>
              <div
                class="ant-col ant-col-lg-3 ant-col-xxl-4"
                style="padding-left:8px;padding-right:8px"
              />
            </div>
            <div
              class="ant-row"
              style="margin-left:-8px;margin-right:-8px;margin-bottom:10px"
            >
              <div
                class="ant-col ant-col-lg-3 ant-col-xxl-2"
                style="padding-left:8px;padding-right:8px"
              >
                <strong>
                  内存(G)：
                </strong>
              </div>
              <div
                class="ant-col ant-col-lg-3 ant-col-xxl-4"
                style="padding-left:8px;padding-right:8px"
              />
              <div
                class="ant-col ant-col-lg-3 ant-col-xxl-2"
                style="padding-left:8px;padding-right:8px"
              >
                <strong>
                  磁盘(G)：
                </strong>
              </div>
              <div
                class="ant-col ant-col-lg-3 ant-col-xxl-4"
                style="padding-left:8px;padding-right:8px"
              >
                 
                <span
                  aria-label="info-circle"
                  class="anticon anticon-info-circle"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    class=""
                    data-icon="info-circle"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                    />
                    <path
                      d="M464 336a48 48 0 1096 0 48 48 0 10-96 0zm72 112h-48c-4.4 0-8 3.6-8 8v272c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V456c0-4.4-3.6-8-8-8z"
                    />
                  </svg>
                </span>
              </div>
            </div>
          </div>
          <h3
            style="border-bottom:1px solid #ececec;padding:10px"
          >
            <span
              aria-label="radar-chart"
              class="anticon anticon-radar-chart"
              role="img"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="radar-chart"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M926.8 397.1l-396-288a31.81 31.81 0 00-37.6 0l-396 288a31.99 31.99 0 00-11.6 35.8l151.3 466a32 32 0 0030.4 22.1h489.5c13.9 0 26.1-8.9 30.4-22.1l151.3-466c4.2-13.2-.5-27.6-11.7-35.8zM838.6 417l-98.5 32-200-144.7V199.9L838.6 417zM466 567.2l-89.1 122.3-55.2-169.2L466 567.2zm-116.3-96.8L484 373.3v140.8l-134.3-43.7zM512 599.2l93.9 128.9H418.1L512 599.2zm28.1-225.9l134.2 97.1L540.1 514V373.3zM558 567.2l144.3-46.9-55.2 169.2L558 567.2zm-74-367.3v104.4L283.9 449l-98.5-32L484 199.9zM169.3 470.8l86.5 28.1 80.4 246.4-53.8 73.9-113.1-348.4zM327.1 853l50.3-69h269.3l50.3 69H327.1zm414.5-33.8l-53.8-73.9 80.4-246.4 86.5-28.1-113.1 348.4z"
                />
              </svg>
            </span>
             CPU
          </h3>
          <div
            style="padding:0 20px 10px 20px"
          >
            <div
              class="ant-row"
              style="margin-left:-8px;margin-right:-8px;margin-bottom:10px"
            >
              <div
                class="ant-col ant-col-lg-3 ant-col-xxl-2"
                style="padding-left:8px;padding-right:8px"
              >
                <strong>
                  主频：
                </strong>
              </div>
              <div
                class="ant-col ant-col-lg-3 ant-col-xxl-4"
                style="padding-left:8px;padding-right:8px"
              />
              <div
                class="ant-col ant-col-lg-3 ant-col-xxl-2"
                style="padding-left:8px;padding-right:8px"
              >
                <strong>
                  核数：
                </strong>
              </div>
              <div
                class="ant-col ant-col-lg-3 ant-col-xxl-4"
                style="padding-left:8px;padding-right:8px"
              />
              <div
                class="ant-col ant-col-lg-3 ant-col-xxl-2"
                style="padding-left:8px;padding-right:8px"
              >
                <strong>
                  供应商：
                </strong>
              </div>
              <div
                class="ant-col ant-col-lg-3 ant-col-xxl-4"
                style="padding-left:8px;padding-right:8px"
              />
              <div
                class="ant-col ant-col-lg-3 ant-col-xxl-2"
                style="padding-left:8px;padding-right:8px"
              >
                <strong>
                  模型：
                </strong>
              </div>
              <div
                class="ant-col ant-col-lg-3 ant-col-xxl-4"
                style="padding-left:8px;padding-right:8px;word-break:break-word"
              />
            </div>
          </div>
          <h3
            style="border-bottom:1px solid #ececec;padding:10px"
          >
            <span
              aria-label="apartment"
              class="anticon anticon-apartment"
              role="img"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="apartment"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M908 640H804V488c0-4.4-3.6-8-8-8H548v-96h108c8.8 0 16-7.2 16-16V80c0-8.8-7.2-16-16-16H368c-8.8 0-16 7.2-16 16v288c0 8.8 7.2 16 16 16h108v96H228c-4.4 0-8 3.6-8 8v152H116c-8.8 0-16 7.2-16 16v288c0 8.8 7.2 16 16 16h288c8.8 0 16-7.2 16-16V656c0-8.8-7.2-16-16-16H292v-88h440v88H620c-8.8 0-16 7.2-16 16v288c0 8.8 7.2 16 16 16h288c8.8 0 16-7.2 16-16V656c0-8.8-7.2-16-16-16zm-564 76v168H176V716h168zm84-408V140h168v168H428zm420 576H680V716h168v168z"
                />
              </svg>
            </span>
             网络信息
          </h3>
          <div
            style="padding:0 20px 10px 20px"
          />
          <h3
            style="border-bottom:1px solid #ececec;padding:10px"
          >
            <span
              aria-label="dashboard"
              class="anticon anticon-dashboard"
              role="img"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="dashboard"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M924.8 385.6a446.7 446.7 0 00-96-142.4 446.7 446.7 0 00-142.4-96C631.1 123.8 572.5 112 512 112s-119.1 11.8-174.4 35.2a446.7 446.7 0 00-142.4 96 446.7 446.7 0 00-96 142.4C75.8 440.9 64 499.5 64 560c0 132.7 58.3 257.7 159.9 343.1l1.7 1.4c5.8 4.8 13.1 7.5 20.6 7.5h531.7c7.5 0 14.8-2.7 20.6-7.5l1.7-1.4C901.7 817.7 960 692.7 960 560c0-60.5-11.9-119.1-35.2-174.4zM761.4 836H262.6A371.12 371.12 0 01140 560c0-99.4 38.7-192.8 109-263 70.3-70.3 163.7-109 263-109 99.4 0 192.8 38.7 263 109 70.3 70.3 109 163.7 109 263 0 105.6-44.5 205.5-122.6 276zM623.5 421.5a8.03 8.03 0 00-11.3 0L527.7 506c-18.7-5-39.4-.2-54.1 14.5a55.95 55.95 0 000 79.2 55.95 55.95 0 0079.2 0 55.87 55.87 0 0014.5-54.1l84.5-84.5c3.1-3.1 3.1-8.2 0-11.3l-28.3-28.3zM490 320h44c4.4 0 8-3.6 8-8v-80c0-4.4-3.6-8-8-8h-44c-4.4 0-8 3.6-8 8v80c0 4.4 3.6 8 8 8zm260 218v44c0 4.4 3.6 8 8 8h80c4.4 0 8-3.6 8-8v-44c0-4.4-3.6-8-8-8h-80c-4.4 0-8 3.6-8 8zm12.7-197.2l-31.1-31.1a8.03 8.03 0 00-11.3 0l-56.6 56.6a8.03 8.03 0 000 11.3l31.1 31.1c3.1 3.1 8.2 3.1 11.3 0l56.6-56.6c3.1-3.1 3.1-8.2 0-11.3zm-458.6-31.1a8.03 8.03 0 00-11.3 0l-31.1 31.1a8.03 8.03 0 000 11.3l56.6 56.6c3.1 3.1 8.2 3.1 11.3 0l31.1-31.1c3.1-3.1 3.1-8.2 0-11.3l-56.6-56.6zM262 530h-80c-4.4 0-8 3.6-8 8v44c0 4.4 3.6 8 8 8h80c4.4 0 8-3.6 8-8v-44c0-4.4-3.6-8-8-8z"
                />
              </svg>
            </span>
             主机负载
          </h3>
          <div
            style="padding:0 20px 10px 20px"
          >
            <div
              class="ant-tabs ant-tabs-top"
            >
              <div
                class="ant-tabs-nav"
                role="tablist"
              >
                <div
                  class="ant-tabs-nav-wrap"
                >
                  <div
                    class="ant-tabs-nav-list"
                    style="transform:translate(0px, 0px)"
                  >
                    <div
                      class="ant-tabs-tab ant-tabs-tab-active"
                    >
                      <div
                        aria-selected="true"
                        class="ant-tabs-tab-btn"
                        role="tab"
                        tabindex="0"
                      >
                        磁盘IO
                      </div>
                    </div>
                    <div
                      class="ant-tabs-tab"
                    >
                      <div
                        aria-selected="false"
                        class="ant-tabs-tab-btn"
                        role="tab"
                        tabindex="0"
                      >
                        网络IO
                      </div>
                    </div>
                    <div
                      class="ant-tabs-tab"
                    >
                      <div
                        aria-selected="false"
                        class="ant-tabs-tab-btn"
                        role="tab"
                        tabindex="0"
                      >
                        内存使用量
                      </div>
                    </div>
                    <div
                      class="ant-tabs-tab"
                    >
                      <div
                        aria-selected="false"
                        class="ant-tabs-tab-btn"
                        role="tab"
                        tabindex="0"
                      >
                        CPU使用率
                      </div>
                    </div>
                    <div
                      class="ant-tabs-tab"
                    >
                      <div
                        aria-selected="false"
                        class="ant-tabs-tab-btn"
                        role="tab"
                        tabindex="0"
                      >
                        根分区使用率
                      </div>
                    </div>
                    <div
                      class="ant-tabs-ink-bar ant-tabs-ink-bar-animated"
                    />
                  </div>
                </div>
                <div
                  class="ant-tabs-nav-operations ant-tabs-nav-operations-hidden"
                >
                  <button
                    aria-controls="null-more-popup"
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    aria-hidden="true"
                    class="ant-tabs-nav-more"
                    id="null-more"
                    style="visibility:hidden;order:1"
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      aria-label="ellipsis"
                      class="anticon anticon-ellipsis"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        class=""
                        data-icon="ellipsis"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
                        />
                      </svg>
                    </span>
                  </button>
                </div>
              </div>
              <div
                class="ant-tabs-content-holder"
              >
                <div
                  class="ant-tabs-content ant-tabs-content-top"
                >
                  <div
                    aria-hidden="false"
                    class="ant-tabs-tabpane ant-tabs-tabpane-active"
                    role="tabpanel"
                    tabindex="0"
                  >
                    <div />
                  </div>
                  <div
                    aria-hidden="true"
                    class="ant-tabs-tabpane"
                    role="tabpanel"
                    style="display:none"
                    tabindex="-1"
                  />
                  <div
                    aria-hidden="true"
                    class="ant-tabs-tabpane"
                    role="tabpanel"
                    style="display:none"
                    tabindex="-1"
                  />
                  <div
                    aria-hidden="true"
                    class="ant-tabs-tabpane"
                    role="tabpanel"
                    style="display:none"
                    tabindex="-1"
                  />
                  <div
                    aria-hidden="true"
                    class="ant-tabs-tabpane"
                    role="tabpanel"
                    style="display:none"
                    tabindex="-1"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </main>
    </section>
  </section>
</section>
`;
